sql: fix RSG test problems

madelynnblue opened the pull request:
- support remaining function paramater types
- prevent transactions from starting and failing

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9373)

<!-- Reviewable:end -->

knz commented on sql/rsg_test.go:
```diff
@@ -108,8 +115,8 @@ func TestRandomSyntaxFunctions(t *testing.T) {
-					i := r.Intn(math.MaxInt64)
-					i -= r.Intn(math.MaxInt64)
+					i := int64(r.Intn(math.MaxInt64))
+					i -= int64(r.Intn(math.MaxInt64))
```
Don't make it so complicated :)
just do `r.Int63()`.

knz commented on sql/rsg_test.go:
```diff
@@ -120,6 +127,24 @@ func TestRandomSyntaxFunctions(t *testing.T) {
+					}
+				case *parser.DDate:
+					i := int64(r.Intn(math.MaxInt64))
+					i -= int64(r.Intn(math.MaxInt64))
```
Ditto

knz reviewed (changes requested):
Don't you also want to exclude syntax using `SAVEPOINT`?

mjibson commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/rsg_test.go, line 119 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9373#-KRe0ft1EA_XsRJR3-2o-r1-119:-KRi2M94bIDLUIc55UZ8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/78d456772e7240c7b2df1047f8d9254ab52df8d9/sql/rsg_test.go#L119)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Don't make it so complicated :)
> 
> just do `r.Int63()`.
> 
> 
</details>
Done.

---

*[sql/rsg_test.go, line 138 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9373#-KRe0ft1EA_XsRJR3-2o-r1-138:-KRi2MtuzHePB_QXmdE-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/78d456772e7240c7b2df1047f8d9254ab52df8d9/sql/rsg_test.go#L138)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Ditto
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9373)*
<!-- Sent from Reviewable.io -->

knz commented:
LGTM

mjibson commented:
The tests passed without ignoring SAVEPOINT, so I think it's ok.

madelynnblue committed:
sql: fix RSG test problems

- support remaining function paramater types
- prevent transactions from starting and failing
